packages/api/src/store/queue.ts

```diff
@@ -137,40 +159,25 @@ export class RabbitQueue implements Queue {
     await this.channel.sendToQueue(QUEUES.events, msg);
   }
 
-  public async publish(route: RoutingKey, msg: messages.Any): Promise<void> {
-    console.log(`publishing to ${route} : ${JSON.stringify(msg)}`);
-    await this.channel.publish(EXCHANGE_NAME, route, msg, { persistent: true });
+  public async publishWebhook(
+    route: RoutingKey,
+    msg: messages.Any
+  ): Promise<void> {
+    console.log(`publishing webhook to ${route} : ${JSON.stringify(msg)}`);
+    await this.channel.publish(EXCHANGES.webhooks, route, msg, {
+      persistent: true,
+    });
   }
 
-  public async delayedPublish(
+  public async delayedPublishWebhook(
     routingKey: RoutingKey,
     msg: messages.Any,
     delay: number
   ): Promise<void> {
-    // TODO: Reimplement this without on-demand queues. Idea: Use a single delayed queue
-    // and per-message expiration and routing key (needs an exchange for that).
-    const delaySec = delay / 1000;
-    const delayedQueueName = `delayed_queue_${routingKey}_${delaySec}s`;
-    const setupFunc = (channel: Channel) =>
-      channel.assertQueue(delayedQueueName, {
-        messageTtl: delay + 100,
-        deadLetterExchange: EXCHANGE_NAME,
-        deadLetterRoutingKey: routingKey,
-        expires: delay + 15000,
-        durable: true,
-      });
-    await this.channel.addSetup(setupFunc);
-    try {
-      console.log(
-        `delayed emitting: delay=${delaySec}s queue=${delayedQueueName} msg=`,
-        msg
-      );
-      await this.channel.sendToQueue(delayedQueueName, msg, {
-        persistent: true,
-      });
-    } finally {
-      // avoid accumulating duplicate setup funcs on the channel manager
-      await this.channel.removeSetup(setupFunc, () => {});
-    }
+    console.log(`emitting delayed message: delay=${delay / 1000}s msg=`, msg);
+    await this.channel.publish(EXCHANGES.delayed, routingKey, msg, {
+      persistent: true,
+      expiration: delay,
```

Comment:
just confused here, `expiration` implies that the event/message would expire after the defined value, whereas `delay` signifies the message would be published after the `value` time has passed. which is it? should it be reworded?

Reply:
to be clear, i get that the channel will delete message after `delay` time has passed and move it to the `deadLetterExchange` declared, my question is more in terms of terminology

Reply:
So it is actually both! One refers to what it means for the delayed publishing "interface", while the other refers to how it is actually implemented in RabbitMQ and thus is passed on the lower-level AMQP publish here.

The pipeline works as this:
 - Message is published to the `delayed` exchange
 - The only queue subscribed to that is the `delayed` queue
 - No one is actually subscribed to that queue, but that queue has a `deadLetterExchange` configured to the regular `webhooks` exchange
 - Once the message reaches its **`expiration`** deadline, it is removed from that queue and re-published on the `webhooks` exchange

So in the low-level RabbitMQ, it really means the **expiration** of the message as the time that the message will spend on the `delayed` queue before being re-published on the main `webhooks` exchange where we bind queues and consume the events.

I think it would be weird to name the public argument as expiration though, as for the high level interface for publishing a delayed message it is just as if the message was delayed by some time before actually reaching the `webhooks` exchange. 

WDYT? Maybe I could add a comment here just above this `publish` or above this `expiration: delay` with some TL;DR of this workflow, to try to avoid confusion.

Reply:
yes, I think a small-ish comment in the function definition should help. When going through the PR, I dived directly through function definition and got confused until i reading PR description

Reply:
Done!